glean-core/android/src/main/java/mozilla/telemetry/glean/Dispatchers.kt

```diff
@@ -28,7 +30,7 @@ internal object Dispatchers {
         private var queueInitialTasks = AtomicBoolean(true)
 
         // Use a [ConcurrentLinkedQueue] to take advantage of it's thread safety and no locking
-        internal val taskQueue: ConcurrentLinkedQueue<suspend CoroutineScope.() -> Unit> = ConcurrentLinkedQueue()
+        internal var taskQueue: TaskQueueType = LinkedList()
```

Comment:
Are you just changing this to the LinkedList in order to be able to insert a task at the head?  I think that's a bad idea as the [documentation for LinkedList](https://developer.android.com/reference/java/util/LinkedList) states that:
> Note that this implementation is not synchronized.

I think that bypassing the queue like @mdboom suggests is a better solution that doesn't require us to use a non-threadsafe list.

Reply:
There is also a [`ConcurrentLinkedDeque`](https://developer.android.com/reference/java/util/concurrent/ConcurrentLinkedDeque) which would allow for inserting at the head of the queue and still makes guarantees about concurrency.

Reply:
> 
> 
> There is also a [`ConcurrentLinkedDeque`](https://developer.android.com/reference/java/util/concurrent/ConcurrentLinkedDeque) which would allow for inserting at the head of the queue and still makes guarantees about concurrency.

Ah, I didn't see that. However, since we're not really using any concurrency feature already (as stated in the commit message), this is hardly of any uses. We're only paying the cost for determining the size of the queue each time we add a new element, as highlighted in their docs 

> `Beware that, unlike in most collections, the size method is NOT a constant-time operation. `

I'm happy with dropping this change, if the `executeTask` works with this as intented.

Reply:
No problem, I'll be glad when this metrics scheduling is all straightened out.